.github/workflows/ci-actions.yml

```diff
@@ -95,6 +95,30 @@ jobs:
       - name: Delete Local Artifacts From Cache
         shell: bash
         run: rm -r ~/.m2/repository/io/quarkus
+      - name: Get GIB arguments
+        id: get-gib-args
+        env:
+          PULL_REQUEST_BASE: ${{ github.event.pull_request.base.sha }}
+        run: |
+          # See also: https://github.com/vackosar/gitflow-incremental-builder#configuration
+          # Common GIB_ARGS for all CI cases (hint: see also pom.xml):
+          # - disableSelectedProjectsHandling: required to detect changes in jobs that use -pl
+          # - untracked: to ignore files created by jobs (and uncommitted to be consistent)
+          GIB_ARGS="-Dgib -Dgib.disableSelectedProjectsHandling -Dgib.untracked=false -Dgib.uncommitted=false"
+          if [ -n "$PULL_REQUEST_BASE" ]
+          then
+              # The PR defines a clear merge target so just use that branch for reference
```

Comment:
When does it happen exactly? I would like to understand when this case happens and when the other case happens?

Typically, I don't understand why we only disable backport PRs in the other case.

Reply:
Also one question related to that: I don't understand exactly how things work for a PR because the remote master might be totally different from the one used to create the PR (if not rebased). I fail to understand how it works in this case?

Reply:
> When does it happen exactly? I would like to understand when this case happens and when the other case happens?

As far I was able to observe/test, this case happens when the workflow run was triggered by a pull request and not just merely by a push to a branch that does not yet have a PR.

> Typically, I don't understand why we only disable backport PRs in the other case.

The idea here is that there is no guessing when you have a PR because the PR clearly defines the target branch. So for some `foo-backport-bar` branch (or whatever) that already has a PR, GIB can simply take the target branch from the PR as reference and compare to that (e.g. `1.10`).

>  I don't understand exactly how things work for a PR because the remote master might be totally different from the one used to create the PR 

When gathering the changes, the source branch (=from where the "working branch", or in GIB terms "base branch", was branched off) is not so important. GIB or better to say JGit will find the common [merge base](https://git-scm.com/docs/git-merge-base) and do the comparison from there.

Btw, I am still hoping that someone with write access to this repo will try my "Idea for testing this closer to reality" (see PR description).
In my fork I wasn't able to properly test with multiple repos. But come to think of it, maybe I should just fork again from my fork to try to simulate it.

Reply:
PS, regarding:
> Typically, I don't understand why we only disable backport PRs in the other case.

I've just corrected the following bit in the PR description which I think contributed to the confusion:

From:
> Note: master, release branches (e.g. 1.10) and backport-PR-branches are not built incrementally (yet)!

To:
> Note: master, release branches (e.g. 1.10) and backport-branches (without PR) are not built incrementally (yet)!

Reply:
PPS:
> But come to think of it, maybe I should just fork again from my fork to try to simulate it.

Hm, doesn't work. You can only have one fork.

Reply:
So... maybe I'm missing something but from what I understood of what you say...

`if [ -n "$PULL_REQUEST_BASE" ]`

We are in the pull request case. We have a proper sha to which we can compare our changes. `PULL_REQUEST_BASE` is not a branch but a sha that points to the commit on top of which we have added the commits in the PR? Right?

If so, in this case, I don't want to use GIB for backport PRs either. I remember having said that before. In the case of the backport PRs, I want a full build in all cases.

`else`

If this case doesn't concern PRs, I would just disable the incremental build altogether. Especially since things are not very clear and we don't have a sha to compare to.

Reply:
> `PULL_REQUEST_BASE` is not a branch but a sha

Yes.

> a sha that points to the commit on top of which we have added the commits in the PR? Right?

Only if e.g. `master` did not receive any new commits since the PR branch was created.
There is a good explanation here: https://github.community/t/how-to-get-base-sha-when-using-pull-request-target/142021/6

> If so, in this case, I don't want to use GIB for backport PRs either.

Personally I think with a PR it is pretty safe already since you have a clear merge target/comparison reference (the most recent SHA at PR creation time from e.g. `1.10`). Additionally, with this PR, `master` and release branches are still built fully.
But these are just my 2 cents and if you want to play it extra safe so be it.

> If this case doesn't concern PRs, I would just disable the incremental build altogether. 

That would mean that branches in forks are always full builds. That's a pretty "hard" decision, IMO.
That's also why I asked on Zulip whether we can always expected "backport" to be contained in names of branch that do not target `master` but a release branch instead (e.g. `1.10`).

Btw, it is not a problem if `master` has advanced since creating the branch because the history of `master` is never rewritten (I suppose).

Reply:
Update: `*.backport.*` branches are now _not_ built incrementally, even if there is a PR.

Reply:
I think is resolved. As of now, branches in forks will also be built incrementally (which I definitely recommend) and `*.backport.*` branches are never built incrementally, with or without PR.